framework/src/modules/chain/submodules/process_transactions.js

```diff
@@ -137,6 +140,56 @@ class ProcessTransactions {
 			transactionsResponses,
 		};
 	}
+
+	// eslint-disable-next-line class-methods-use-this
+	async processSignature(signature) {
+		// Get data required for processing signature
+		const stateStore = library.logic.stateManager.createStore({
+			mutate: false,
+		});
+
+		// Grab transaction with corresponding ID from transaction pool
+		const transaction = modules.transactions.getMultisignatureTransaction(
+			signature.transactionId
+		);
+		if (!transaction) {
+			const message = `Unable to process signature, corresponding transaction not found: ${
+				signature.transactionId
+			}`;
+			library.logger.error(message, { signature });
+			return {
+				id: signature.transactionId,
+				status: TransactionStatus.FAIL,
+				errors: [new Error(message)],
+			};
+		}
+
+		// Add multisignature to transaction and process
+		const transactionResponse = transaction.addMultisignature(
+			stateStore,
+			signature
+		);
+		if (transactionResponse.status === TransactionStatus.FAIL) {
+			library.logger.error(transactionResponse.errors[0].message, {
+				signature,
+				transaction: transaction.toJSON(),
+			});
+		}
+
+		return transactionResponse;
+	}
 }
 
+// Events
+/**
+ * Assigns modules to private constant `modules`.
+ *
+ * @param {modules} scope - Loaded modules
+ */
+ProcessTransactions.prototype.onBind = function(scope) {
```

Comment:
`ProcessTransactions` has no dependencies on other modules and it doesn't change the state of the application or the transaction pool. I like it this way and I think we should stick to it.
I think we can follow the existing code flow structure in which the transaction is received in the `transport` module, which internally calls `multisignature` module where transaction signatures are processed. 
Therefore, we should define the logic for fetching the state for the transaction and adding the signature in processSignatrure function but the transaction should be fetched from the transaction pool in the `multisignature` module.